File: Solutions/MoNaDDu/Game Shop/1/Part1.hs
module Part1 where

data Item = Item { name :: String, description :: String, weight :: Int, value :: Int }
 deriving ( Show )

type Items = [Item]

testCase1 = describe items

describe :: Items -> IO ()
describe = mapM_ (putStrLn . toString)

toString :: Item -> String
toString item = "Name        = " ++ name item ++ "\n" ++
                "Description = " ++ description item ++ "\n" ++
                "Weight      = " ++ w'' ++ "\n" ++
                "value       = " ++ v'' ++ "\n"
                where 
                 w = weight item
                 v = value item
                 w' = show w
                 v' = show v
                 w''
                  | w > 1 = w' ++ " lbs"
                  | otherwise = w' ++ " lb"
                 v''
                  | v > 1 = v' ++ " gold coins"
                  | otherwise = v' ++ " gold coin"

items = [Item { name = "Excalibur", description = "The legendary sword of King Arthur", weight = 12, value = 1024 }, Item { name = "Steel Armor", description = "Protective covering made by steel", weight = 15, value = 805 }]

File: Solutions/Jinho1011/Game Shop/Part 1.java
package test;

class Item {
    private String name;
    private String description;
    private int weight;
    private int value;

    public Item(String name, String description, int weight, int value) {
        this.name = name;
        this.description = description;
        this.weight = weight;
        this.value = value;
    }

    public void Describe() {
        System.out.println("Name        = "+name+"");
        System.out.println("Description = "+description+"");
        System.out.println("Weight      = "+weight+" lbs");
        System.out.println("Value       = "+value+" gold coins\n");
    }
}

class ItemTest {
    public  static void main (String args []) {
        Item weapon, armor;
        weapon = new Item ("Excalibur", "The legendary sword of King Arthur",12,1024);
        weapon.Describe();


        armor = new Item ("Steel Armor","Protective covering made by steel",15,805);
        armor.Describe();
    }
}
